### PR TITLE
WG Naming: Add language evaluation framework

### PR DESCRIPTION
**Pull request by celestehorgan**
Signed-off-by: Celeste Horgan <celeste@cncf.io>

Add the first version of the language evaluation framework.

### CODE REVIEW

**Comment by celestehorgan** [comment]
cc: @justaugustus @zacharysarah @jdumars

File: wg-naming/language-evaluation-framework.md
Comment (ricochet) [inline wg-naming/language-evaluation-framework.md:1]: extra newline

File: wg-naming/language-evaluation-framework.md
Comment (ricochet) [inline wg-naming/language-evaluation-framework.md:8]: outlined*

File: wg-naming/language-evaluation-framework.md
Comment (ricochet) [inline wg-naming/language-evaluation-framework.md:62]: extra newline

File: wg-naming/language-evaluation-framework.md
Comment (ricochet) [inline wg-naming/language-evaluation-framework.md:70]: extra newline

File: wg-naming/language-evaluation-framework.md
Comment (ricochet) [inline wg-naming/language-evaluation-framework.md:20]: extra newline

File: wg-naming/language-evaluation-framework.md
Comment (ricochet) [inline wg-naming/language-evaluation-framework.md:47]: socioeconomic status is the only category that I see as potentially missing.

Examples include adjectives that label people “the poor” vs specific designations.

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:2]:
```suggestion
# A framework for evaluating harmful language
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:8]:
```suggestion
The framework was created for an open source technology project. The framework may be applicable to other fields as well.
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:12]:
```suggestion
## Using the framework
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:14]:
```suggestion
The framework is divided into three sections: first-, second-, and third-order concerns, ranked in order of potential harm to the community. 
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:16]:
```suggestion
First-order concerns are language where harm is egregious, overt, and clearly problematic. Second-order concerns are language which is problematic but with a less definite impact. Third-order concerns indicate language that could use improvement but does no demonstrable harm.
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:18]:
```suggestion
Answer all questions for each term evaluated.
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:21]:
```suggestion
When complete, consider questions answered in the affirmative: in general, the more questions answered “yes” or “possibly”, the more likely it is that the language in question needs to be replaced. 

If any first-order concerns are a “yes”, replace the language. 

If a significant number of second- or third- order concerns are a “yes”, strongly consider replacing the language. 
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:23]:
```suggestion
This framework is intentionally non-prescriptive. The intention in this work is to reduce harm for the community; let harm reduction guide your decisions.
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:29]: Punctuate bullets consistently.
```suggestion
- Overtness: regardless of its use in the context of code or technology, there is little to no ambiguity outside of technology as to whether the language in question indicates harm
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:30]:
```suggestion
- Identity-specificity: language in question specifically unambiguously identifies a group of people
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:53]:
```suggestion
- Lack of specific identity: concerns in this category do not target specific identities, or do so in a non-overt way
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:68]:
```suggestion
- Clarity: is the language in use a metaphor that could be described more precisely using different words?

- Anthropomorphism: does language unnecessarily humanize components or processes?

- Idiomatic: Is language unclear to someone outside a specific culture?
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:81]:
```suggestion
### Changes over time
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:85]:
```suggestion
As a result, terms that were once deemed acceptable may, at some future point, be deemed unacceptable. 
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:91]:
```suggestion
- Openness to updating language as readers and cultures change
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:96]:
```suggestion
In the handful of months since this work began, both Kubernetes as a whole and WG Naming have dealt with a number of issues and comments from trolls. We anticipate that anyone using this document to guide their own work will receive the same kind of attention. 
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:98]:
```suggestion
In Kubernetes we mostly encounter [sea lions](http://wondermark.com/1k62/) (concern trolls), who seek to legitimize debate over false concerns in order to use up contributors' energy and time.
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:100]:
```suggestion
We work with our GitHub and other moderation teams to shut down trolling behavior at the source and remove trolling content. 
```

File: wg-naming/language-evaluation-framework.md
Comment (zacharysarah) [inline wg-naming/language-evaluation-framework.md:103]:
```suggestion

Rather than be discouraged by trolls, consider it a heartening sign that you are engaged in meaningful work. 
```

**Review by zacharysarah — CHANGES_REQUESTED** [review]
@celestehorgan Thanks for drafting this. I've made some revisions, some of which are substantial.

File: wg-naming/language-evaluation-framework.md
Comment (celestehorgan) [reply wg-naming/language-evaluation-framework.md:68]: I see this :+1:

**Review by spiffxp — APPROVED** [review]
/hold
to be removed at wg-naming approvers' discretion

/lgtm
I didn't have any detailed comments to add.  I found this useful to reference in a PR recommending alternatives to 'master' https://github.com/kubernetes/community/pull/5243

**Comment by justaugustus** [comment]
Great work, @celestehorgan, and thanks to everyone for the reviews!
/lgtm
/approve
/hold cancel

**Pull request merged**
